stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -182,6 +182,59 @@ for elty in (Float32,Float64)
     end
 end
 
+"""
+    muladd(A, y, z)
+
+Combined multiply-add, `A*y .+ z`, for either matrix-matrix multiplication or
+matrix-vector multiplication. The result is always the size of `A*y`, although
+`z` may have fewer dimensions.
+
+!!! compat "Julia 1.6"
+     These methods require Julia 1.6 or later.
+
+# Examples
+```jldoctest
+julia> A=[1.0 2.0; 3.0 4.0]; B=[1.0 1.0; 1.0 1.0]; C=[0, 100];
+
+julia> muladd(A, B, C)
+2×2 Matrix{Float64}:
+   3.0    3.0
+ 107.0  107.0
+```
+"""
+function Base.muladd(A::AbstractMatrix{TA}, y::AbstractVector{Ty}, z) where {TA, Ty}
+    T = promote_type(TA, Ty, z isa AbstractArray ? eltype(z) : typeof(z))
```

Comment:
Which cases do you have in mind for which `eltype` doesn't work? For `Number`s, at least, it does.

Reply:
I was imagining arrays of matrices, e.g. I think many things in LinearAlgebra work if, instead of complex numbers, you use as elements 2x2 matrices.

Reply:
Wait, but arrays of matrices fall in the first (i.e., the `eltype`) category, right? I was wondering if you can't simply call `eltype` on any type of `z` you can imagine. Or conversely, for which type of `z` do you think you need `typeof(z)`?

Reply:
Oh you might be right, sorry. My pauli matrices would go wrong in `A*x .+ z` which `muladd` is supposed to mimmic, an on the next line’s broadcast too.

Reply:
In any case, if you imagine such a use case, I think we should nail the intended behavior with a test.

Reply:
I've now added some vector-of-matrices tests. One fails I think because of a 5-arg mul problem but I need to look a bit more. 

Besides `eltype(::Number)`, it's safe to use `ndims(::Number)==0` right?

Reply:
Yes, there is
```julia
julia> @which ndims(3.)
ndims(x::Number) in Base at number.jl:67
```